modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBMessageLocalServiceImpl.java

```diff
@@ -2362,8 +2362,19 @@ private String _getUniqueUrlSubject(
 			urlSubject = urlSubject + StringPool.DASH;
 		}
 
+		int maxLength = ModelHintsUtil.getMaxLength(
+			MBMessage.class.getName(), "urlSubject");
+
 		for (int i = 1; mbMessage != null; i++) {
-			uniqueUrlSubject = urlSubject + i;
+			String suffix = StringPool.DASH + i;
+
+			urlSubject = (urlSubject.length() > (maxLength - suffix.length())) ?
```

Comment:
On liferay we (sadly) do not use the ternary on java, so please change to a if / else instead